Sources/Subscription/DeadTokenRecoverer.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  DeadTokenRecoverer.swift
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Networking
+import os.log
+
+@available(macOS 12.0, *)
+public struct DeadTokenRecoverer {
+
+    private static var recoveryAttempted: Int = 0
```

Comment:
In case the BE has a problem and always returns a `OAuthClientError.invalidTokenRequest` every time we request a token, we don't want to enter an infinite loop, so we must to be sure that we are not already in a recovery state.
This state helps with that and it's not a bool because multiple, parallel API requests could use this utility and we want to be sure we don't have concurrency issues.